src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -1,4 +1,17 @@
+﻿@using Microsoft.Extensions.Configuration
+@using System.Globalization
+@using OrchardCore.Environment.Shell.Configuration
+@inject Microsoft.AspNetCore.Mvc.Localization.IViewLocalizer T
 @inject OrchardCore.IOrchardHelper Orchard
+@inject IShellConfiguration ShellConfiguration
+
+@{
+    var configurationSection = ShellConfiguration.GetSection("OrchardCore.Setup");
+    var currentCulture = Orchard.HttpContext.Request.Query.ContainsKey("ui-culture")
```

Comment:
Can't we use the default culture providers to pick the correct culture, and select the drop down based on the "CurrentCulture" ?

Reply:
Probably we can